docs/src/architecture/assets.md

```diff
@@ -0,0 +1,57 @@
+# Assets
+The goal of the Miden Rollup is to provide a reliable protocol and software for users to exchange assets. In Miden, users can create and exchange arbitrary fungible and non-fungible assets. 
+
+The way how assets are recorded in Polygon Miden suffices four goals
+
+* Asset exchange should be parallelizable 
+* Asset ownership should be private 
+* Asset usage should be truly censorship resistant
+* Fees can be paid using any asset
+
+## Definition
+Assets are data structures that can be stored in account and note vaults. In Miden, there are fungible and non-fungible assets. All assets are encoded using a single `Word` (4 elements) such that it is easy to determine the type of an asset both inside and outside Miden VM. 
+
+Specifically, the first element will be
+
+   * `ZERO` for a fungible asset: `[e0, ZERO, e2, e3]`
+   * `non-ZERO` for a non-fungible asset: `[e0, non-ZERO, e2, e3]`
+  
+and the most significant bit will be
+
+   * `ONE` for a fungible asset
+   * `ZERO` for a non-fungible asset
```

Comment:
I'm not sure we need to go into that much detail - this feels more like specs rather than higher level description.

Reply:
I think you are right